app/controllers/assessments_controller.rb

```diff
@@ -137,7 +137,6 @@ def update
     redirect_to(root_url) && return if patient.nil?
 
     redirect_to root_url unless current_user&.can_edit_patient_assessments?
-    patient = Patient.find_by(submission_token: params.permit(:patient_submission_token)[:patient_submission_token])
```

Comment:
A few lines above this we already had found the patient using the same criteria.